daemon/list.go

```diff
@@ -137,6 +138,14 @@ func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container,
 			Names: names[container.ID],
 		}
 		newC.Image = container.Config.Image
+		if container.Config.Image != "" {
+			img, err := daemon.repositories.LookupImage(container.Config.Image)
```

Comment:
And this change is also to make the client side easy to check if the image identifier is a name or ID.